src/heroes/sombra/translocator_cooldown_reset.opy

```diff
@@ -0,0 +1,64 @@
+#Global variables
+
+globalvar STEALTH_REFUND_FRACTION 0
+globalvar NEGATIVE_STATUS_PREVENTS_REFUND 1
+globalvar MINIMUM_TRANSLOCATOR_COOLDOWN 2
+
+
+#Player variables
+
+playervar stealth_timer 0
+playervar stealth_max_Cooldown 1
+playervar ult_charge_Percent_cache 2
+
+
+rule "Initial Global":
+    STEALTH_REFUND_FRACTION = createWorkshopSetting(int[0:100], "Gameplay Settings", "Percent of Stealth Timer Remaining Refunded Towards Translocator Cooldown", 40, 0) / 100
+    NEGATIVE_STATUS_PREVENTS_REFUND = createWorkshopSetting(bool, "Gameplay Settings", "Negative Status Effects Cancelling Stealth Prevents Translocator Cooldown Refund", true, 1)
+    MINIMUM_TRANSLOCATOR_COOLDOWN = 4
+    
+    
+rule "When Translocator is used, begin tracking how long Sombra is in stealth":
+    @Event eachPlayer
+    @Condition (eventPlayer.getCurrentHero() == Hero.SOMBRA or eventPlayer.getHeroOfDuplication() == Hero.SOMBRA) == true
```

Comment:
Generally, echo duplicate shouldn't be supported in order to prevent lag

Reply:
I have removed the check for echo.